### ISSUE TITLE
lightgbm 3.3.1 brings LGBMNotFittedError

### ISSUE THREAD

**Issue by jyker** [mandiant/stringsifter#18]
- system
ubuntu-20.04 + stringsifter-2.20201202

- issue
rank_strings meets LGBMNotFittedError when lightgbm >= 3.3.1, last work version: lightgbm == 3.3.0

```shell
test@test:/dist# flarestrings -n 8 ./main | rank_strings -l 5
Traceback (most recent call last):
  File "/usr/local/bin/rank_strings", line 8, in <module>
    sys.exit(argmain())
  File "/usr/local/lib/python3.8/site-packages/stringsifter/rank_strings.py", line 140, in argmain
    main(args.input_strings, args.limit, args.min_score,
  File "/usr/local/lib/python3.8/site-packages/stringsifter/rank_strings.py", line 39, in main
    y_scores = ranker.predict(X_test)
  File "/usr/local/lib/python3.8/site-packages/lightgbm/sklearn.py", line 795, in predict
    raise LGBMNotFittedError("Estimator not fitted, call fit before exploiting the model.")
sklearn.exceptions.NotFittedError: Estimator not fitted, call fit before exploiting the model.
```

**Comment by phantinuss** [comment]
Same issue. Manually setting lightbgm to 3.3.0 fixed the issue for me. New package installations of stringsifter will have this issue. If not fixed lightgbm could at least be pinned to 3.3.0, couldn't it?

**Comment by dmknght** [comment]
This error is happening on my side. Luckily install lightgbm 3.3.0 fixed it (`sudo pip3 install lightgbm==3.3.0`)

**Comment by ewalshmndt** [comment]
Fixed by #34
